lms/djangoapps/lti_provider/tasks.py

```diff
@@ -0,0 +1,92 @@
+"""
+Asynchronous tasks for the LTI provider app.
+"""
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.dispatch import receiver
+import logging
+
+from courseware.models import SCORE_CHANGED
+from lms import CELERY_APP
+from lti_provider.models import GradedAssignment
+import lti_provider.outcomes
+from lti_provider.views import parse_course_and_usage_keys
+
+log = logging.getLogger("edx.lti_provider")
+
+
+@receiver(SCORE_CHANGED)
+def score_changed_handler(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Consume signals that indicate score changes. See the definition of
+    courseware.models.SCORE_CHANGED for a description of the signal.
+    """
+    points_possible = kwargs.get('points_possible', None)
+    points_earned = kwargs.get('points_earned', None)
+    user_id = kwargs.get('user_id', None)
+    course_id = kwargs.get('course_id', None)
+    usage_id = kwargs.get('usage_id', None)
+
+    if None not in (points_earned, points_possible, user_id, course_id, user_id):
+        lti_provider.tasks.send_outcome.delay(
+            points_possible,
+            points_earned,
+            user_id,
+            course_id,
+            usage_id
+        )
+    else:
+        log.error(
+            "Outcome Service: Required signal parameter is None. "
+            "points_possible: %s, points_earned: %s, user_id: %s, "
+            "course_id: %s, usage_id: %s",
+            points_possible, points_earned, user_id, course_id, usage_id
+        )
+
+
+@CELERY_APP.task
+def send_outcome(points_possible, points_earned, user_id, course_id, usage_id):
+    """
+    Calculate the score for a given user in a problem and send it to the
+    appropriate LTI consumer's outcome service.
+    """
+    course_key, usage_key = parse_course_and_usage_keys(course_id, usage_id)
+    try:
+        assignment = GradedAssignment.objects.get(
+            user=user_id, course_key=course_key, usage_key=usage_key
+        )
+    except ObjectDoesNotExist:
```

Comment:
The plan is to send to both of them. This is the corner case that I talked about in an earlier comment; I have code handling it in a later change, rather than complicating this one. I'll pull that code into this change.